### DISCUSSION TITLE
Renovate not working for private artifactory

### DISCUSSION THREAD

**Discussion by mrVragec** [renovatebot/renovate#12499] [Request Help]
Hello guys, 

in the company, we use Renovate for a lot of projects for more than a year, unfortunately, Renovate stopped working (for our internal artifactory) after (around) 30th August. It's still working for the public maven repository.
We didn't change anything in our configuration :( 
Our projects are Kotlin with Gradle, hosted on GitHub, and Renovate is used as a GitHub app.

Could someone point me in the correct direction, to fix this issue? 

This is our configuration: 
In each of our repositories, we have **renovate.json** with this content.
```
{
    "extends": [
        "github><xxx>/renovate-config"
    ],
    "reviewers": [
        "team:<xxx>/team"
    ]
}
```

The base config is in **default.json** in <xxx>/renovate-config and this is the content of the file
```
{
  "extends": [
    "config:base"
  ],
  "timezone": "Europe/Berlin",
  "maven": {
    "enabled": true
  },
  "hostRules": [
    {
      "matchHost": "azure.com",
      "username": "<redacted>",
      "encrypted": {
        "password": "<redacted>"
      }
    }
  ],
  "packageRules": [
    {
      "registryUrls": [
        "https://pkgs.dev.azure.com/<xxx>/<abc>/_packaging/<abc>/maven/v1",
        "https://pkgs.dev.azure.com/<xxx>/<abc>/_packaging/<bcd>/maven/v1"
      ]
    },
    {
      "matchUpdateTypes": [
        "minor",
        "patch",
        "pin",
        "digest"
      ],
      "automerge": true
    }
  ]
}
```

**Comment by rarkins** [comment]
1. Did you mean to say "internal registry" instead of "internal artifactory"? You appear to be pointing to Azure DevOps and not JFrog Artifactory.
2. If you log into app.renovatebot.com and look at a recent job log, do you see logs noting lookup errors?
3. If you find the "packageFiles with updates" log message, are the internal dependencies listed there at all, or only the public ones?

**Comment by mrVragec** [comment]
Hi @rarkins 
1. yes, sorry for my mistake
2. there are a lot of entries, for our internal libs, in logs with "Content is not found for Maven URL", but there are no entries with ERROR level or even no "error" string in the log
3. yes, our internal dependencies are listed there but with this message _"message": "Failed to look up dependency ****"_ and in the "registryUrls" section for the particular internal dependency there is only a public maven repository, our internal registry is not there. 
For mee looks like the internal registry somehow is not detected or recognized, could it be?

**Reply by rarkins** [reply]
I spotted a mistake in the packageRules. Please try this instead:

```
{
  "extends": [
    "config:base"
  ],
  "timezone": "Europe/Berlin",
  "hostRules": [
    {
      "matchHost": "azure.com",
      "username": "<redacted>",
      "encrypted": {
        "password": "<redacted>"
      }
    }
  ],
  "packageRules": [
    {
      "matchDatasources": ["maven"],
      "registryUrls": [
        "https://pkgs.dev.azure.com/<xxx>/<abc>/_packaging/<abc>/maven/v1",
        "https://pkgs.dev.azure.com/<xxx>/<abc>/_packaging/<bcd>/maven/v1"
      ]
    },
    {
      "matchUpdateTypes": [
        "minor",
        "patch",
        "pin",
        "digest"
      ],
      "automerge": true
    }
  ]
}
```

Note the addition of `"matchDatasources": ["maven"],` in the first rule. I also removed the unnecessary maven.enabled=true config although it shouldn't be causing any problems.

**Reply by mrVragec** [reply]
@rarkins Works like a charm. I don't know why did not work before, as our config was almost the same only the "matchDatasources": ["maven"] was missing. But as mentioned, in the past worked also without this. 
Thank you !!
